src/XMakeBuildEngine/Evaluation/Expander.cs

```diff
@@ -253,6 +253,7 @@ internal string ExpandIntoStringLeaveEscaped(string expression, ExpanderOptions
             string result = MetadataExpander.ExpandMetadataLeaveEscaped(expression, _metadata, options);
             result = PropertyExpander<P>.ExpandPropertiesLeaveEscaped(result, _properties, options, elementLocation, _usedUninitializedProperties);
             result = ItemExpander.ExpandItemVectorsIntoString<I>(this, result, _items, options, elementLocation);
+            result = FileUtilities.MaybeAdjustFilePath(result);
 
```

Comment:
Does this work if you have e.g. an expression that expands to something like "a\b\c.txt;d\e\f.txt" ?  Perhaps add some unit tests of the function itself?